src/render.rs

```diff
@@ -54,6 +56,22 @@ pub const fn next_multiple_of(val: u32, rhs: u32) -> u32 {
     }
 }
 
+// This must be kept in sync with the struct in shaders/shared/bump.wgsl
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, Zeroable, Pod)]
+struct BumpAllocators {
+    failed: u32,
+    binning_size: u32,
+    ptcl_size: u32,
+    tiles_size: u32,
+    segments_size: u32,
```

Comment:
Also, is there a reason these aren't just measured using `arrayLength` in the shaders instead? I guess it's useful to keep the data around CPU side.

```suggestion
    // Sizes of the provided buffers
    binning_size: u32,
    ptcl_size: u32,
    tiles_size: u32,
    segments_size: u32,
    // Final needed dynamic size of the buffers. If any of these are larger than the corresponding `_size` element
    // reallocation needs to occur
```

Reply:
I'll add a note here, hopefully one of us will remember at the time it's relevant:

It would be short-sighted to set the corresponding `_size` to exactly this requested value, because we round buffer sizes up in `engine`; this should be set based on the the new rounded sizes instead.

Reply:
I’m reluctant to use `arrayLength` because it’s not supported on Metal without contrivances and I’d prefer not to have to deal with those in a future native backend.